SparseGrids/TasmanianSparseGrid.cpp

```diff
@@ -479,15 +493,31 @@ void TasmanianSparseGrid::integrate(double q[]) const{
 }
 
 void TasmanianSparseGrid::differentiate(const double x[], double jacobian[]) const {
+    // For a grid with a transformed-to-canonical operator g(x) and model f(x) over the canonical domain, this returns the
+    // Jacobian of f(g(x)), i.e., [Jacobian of f(.) at g(x)] * [Jacobian of g(.) at x].
+    Data2D<double> x_tmp;
+    // Jacobian of f(.) at g(x).
     if (isGlobal()) {
-        get<GridGlobal>()->differentiate(x, jacobian);
+        get<GridGlobal>()->differentiate(formCanonicalPoints(x, x_tmp, 1), jacobian);
     } else if (isSequence()) {
-        get<GridSequence>()->differentiate(x, jacobian);
+        get<GridSequence>()->differentiate(formCanonicalPoints(x, x_tmp, 1), jacobian);
     } else if (isLocalPolynomial()) {
-        get<GridLocalPolynomial>()->differentiate(x, jacobian);
+        get<GridLocalPolynomial>()->differentiate(formCanonicalPoints(x, x_tmp, 1), jacobian);
+    } else if (isFourier()) {
+        get<GridFourier>()->differentiate(formCanonicalPoints(x, x_tmp, 1), jacobian);
     } else {
         throw std::runtime_error("ERROR: in differentiate(), jacobians/gradients are not available for this type of grid");
     }
+    // Jacobian of f(g(.)) at x.
+    if (domain_transform_a.size() != 0 or conformal_asin_power.size() != 0) {
```

Comment:
Same here, throw and exception if a transform has been set.

Reply:
Same as above